File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.swift
import Foundation

func solution(_ numer1:Int, _ denom1:Int, _ numer2:Int, _ denom2:Int) -> [Int] {
    let numer3 = (numer1 * denom2) + (numer2 * denom1)
    let denom3 = (denom1 * denom2)
    
    let gcd = (numer3 > denom3) ? getGCD(numer3, denom3) : getGCD(denom3, numer3)
    
    return [(numer3 / gcd), (denom3 / gcd)]
}

func getGCD(_ x:Int , _ y:Int) -> Int {
    // a > b
    var a = x
    var b = y
    var r = a % b
    
    while r != 0 {
        a = b
        b = r
        r = a % b
    }
    
    return b
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.cpp
#include <algorithm>
#include <iostream>
#include <queue>
#include <string>
#include <vector>

using namespace std;

int getGCD(int a, int b) {
    int c = a % b;

    while (c != 0) {
        a = b;
        b = c;
        c = a % b;
    }

    return b;
}

vector<int> solution(int numer1, int denom1, int numer2, int denom2) {
    vector<int> answer;

    int x, y, GCD;

    y = denom1 * denom2;
    x = numer1 * denom2 + numer2 * denom1;

    GCD = x > y ? getGCD(x, y) : getGCD(y, x);

    answer.push_back(x / GCD);
    answer.push_back(y / GCD);

    return answer;
}